jerry-core/jerryscript.h

```diff
@@ -322,9 +335,19 @@ jerry_value_t jerry_get_object_keys (const jerry_value_t obj_val);
 jerry_value_t jerry_get_prototype (const jerry_value_t obj_val);
 jerry_value_t jerry_set_prototype (const jerry_value_t obj_val, const jerry_value_t proto_obj_val);
 
+__attr_deprecated___
 bool jerry_get_object_native_handle (const jerry_value_t obj_val, uintptr_t *out_handle_p);
+__attr_deprecated___
 void jerry_set_object_native_handle (const jerry_value_t obj_val, uintptr_t handle_p,
                                      jerry_object_free_callback_t freecb_p);
+
+bool jerry_get_object_native_pointer (const jerry_value_t obj_val,
+                                      uintptr_t *out_native_p,
+                                      const jerry_object_native_info_t **out_info_p);
+void jerry_set_object_native_pointer (const jerry_value_t obj_val,
+                                      uintptr_t native_p,
```

Comment:
Maybe we can get rid of the jerry_dispatch_object_free_callback, and directly call the free_cb in the `ecma_gc_free_native_pointer`. Because in ecma-gc can get access to jerryscript.h